packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -177,6 +186,75 @@ export function MediaPlaceholder( {
 		} );
 	};
 
+	async function handleBlocksDrop( blocks ) {
+		if ( ! blocks || ! Array.isArray( blocks ) ) {
+			return;
+		}
+
+		function recursivelyFindImagesFromBlocks( _blocks ) {
+			return _blocks.flatMap( ( block ) =>
+				block.name === 'core/image'
```

Comment:
Should we check that the image is actually "filled" (like drag and dropping empty image blocks shouldn't do anything)